src/assert.rs

```diff
@@ -356,28 +360,31 @@ impl Assert {
         let output = spawned.wait_with_output()?;
 
         if let Some(expect_success) = self.expect_success {
-            if expect_success != output.status.success() {
-                let out = String::from_utf8_lossy(&output.stdout).to_string();
-                let err = String::from_utf8_lossy(&output.stderr).to_string();
-                let err: Error = ErrorKind::StatusMismatch(expect_success, out, err).into();
-                bail!(err.chain_err(|| ErrorKind::AssertionFailed(self.cmd.clone())));
+            let actual_success = output.status.success();
+            if expect_success != actual_success {
+                Err(failure::Context::new(StatusError::new(
+                    actual_success,
+                    output.stdout.clone(),
+                    output.stderr.clone(),
+                )).context(AssertionError::new(self.cmd.clone())))?;
```

Comment:
This was a concern a brought up in a different section on the previous commit but that got resolved by the change and `bail!`.  Here it is still relevant.

> I used `?` to turn a `Context` into an `Error`.  I felt dirty with it being hard to see in this case.  I somewhat prefer falling back to the old `try!` macro here.

> The `throw!` macro that is planned to be added will help with this.

> I wonder if `fail` should `impl From<Result<T, Context<D>>> for for Result<T, Error>`. so I could do

> return Err(AssertionKind).into()

> or maybe `impl From<Fail> for for Result<T, Error>`

> return AssertionKind.into()